src/meta/src/hummock/manager/mod.rs

```diff
@@ -1160,85 +1266,127 @@ where
                 level_deltas.push(LevelDelta {
                     delta_type: Some(DeltaType::GroupConstruct(GroupConstruct {
-                        group_config: Some(compaction_config),
+                        group_config: Some(compaction_config.clone()),
                     })),
                 });
             }
         }
+
+        if !need_independent_commit {
+            return Ok((
+                new_version_delta.key,
+                new_version_delta.new_value,
+                new_hummock_version,
+            ));
+        }
+
+        new_version_delta.max_committed_epoch = new_hummock_version.max_committed_epoch;
+        commit_multi_var!(self, None, new_version_delta)?;
+        branched_ssts.commit();
+        versioning.current_version = new_hummock_version;
+
+        self.env
+            .notification_manager()
+            .notify_hummock_asynchronously(
+                Operation::Add,
+                Info::HummockVersionDeltas(risingwave_pb::hummock::HummockVersionDeltas {
+                    version_deltas: vec![versioning
+                        .hummock_version_deltas
+                        .last_key_value()
+                        .unwrap()
+                        .1
+                        .clone()],
+                }),
+            );
+
+        old_version = versioning.current_version.clone();
+        new_version_id = old_version.id + 1;
+        let mut new_version_delta1 = BTreeMapEntryTransaction::new_insert(
+            &mut versioning.hummock_version_deltas,
+            new_version_id,
+            HummockVersionDelta {
+                prev_id: old_version.id,
+                safe_epoch: old_version.safe_epoch,
+                trivial_move: false,
+                ..Default::default()
+            },
+        );
+
+        new_hummock_version = old_version;
+        new_version_delta1.id = new_version_id;
+        new_hummock_version.id = new_version_id;
+        Ok((
+            new_version_delta1.key,
+            new_version_delta1.new_value,
+            new_hummock_version,
+        ))
     }
 
     /// Caller should ensure `epoch` > `max_committed_epoch`
     #[named]
     pub async fn commit_epoch(
         &self,
         epoch: HummockEpoch,
-        sstables: Vec<LocalSstableInfo>,
+        mut sstables: Vec<LocalSstableInfo>,
         sst_to_context: HashMap<HummockSstableId, HummockContextId>,
     ) -> Result<()> {
+        let mut compaction_guard = write_lock!(self, compaction).await;
         let mut versioning_guard = write_lock!(self, versioning).await;
         let _timer = start_measure_real_process_timer!(self);
         // Prevent commit new epochs if this flag is set
         if versioning_guard.disable_commit_epochs {
             return Ok(());
         }
 
-        let compaction_groups: HashMap<_, _> = self
+        let mut compaction_groups: HashMap<_, _> = self
             .compaction_group_manager
             .compaction_groups()
             .await
```

Comment:
> May not be related to this PR: we should add an assertion somewhere to ensure that exclusive group assignment for any table id, in other words, a table id is only assigned to one compaction group.

checked in my latest commit

Reply:
> May not be related to this PR: we should add an assertion somewhere to ensure that exclusive group assignment for any table id, in other words, a table id is only assigned to one compaction group.

done in https://github.com/risingwavelabs/risingwave/pull/5792/commits/c3103bea69681ca42ad41961d6a0826eb8cfdbc1

Reply:
> May not be related to this PR: we should add an assertion somewhere to ensure that exclusive group assignment for any table id, in other words, a table id is only assigned to one compaction group.

prior check to prevent this situation (not assertion after this situation) is done in https://github.com/risingwavelabs/risingwave/pull/5792/commits/e6946d8b57abb5fba36da9cf70cab275fc65f696